### ISSUE TITLE
Performance: every IPC call spawns a new detached thread instead of using a thread pool

### ISSUE THREAD

**Issue by KrisPowers** [bindertools/binder#39]
## Summary

In `cpp/host/dispatch.cpp` (lines 241-251), every non-urgent IPC call creates a brand new `std::thread` and detaches it:

```cpp
std::thread([this, seq, type, args]() {
    try {
        json args_json = args.empty() ? json::object() : json::parse(args);
        dispatch_worker(seq, type, args_json);
    } catch (const std::exception& e) {
        resolve_err(seq, std::string("dispatch error: ") + e.what());
    }
}).detach();
```

This means:
- Under load (rapid file opens, git ops, search queries), a new OS thread is created for every keystroke-triggered IPC call.
- There is no upper bound on concurrent threads. A slow backend call (like a large directory scan) creates a thread that can stay alive for seconds while new requests continue to create more threads.
- Detached threads cannot be joined or cancelled on shutdown, which can cause use-after-free issues with `this` (the Dispatcher) after it is destroyed.

## How to fix

Replace the detach pattern with a fixed-size thread pool. The simplest approach is a `std::vector<std::thread>` with a `std::deque` task queue protected by a `std::mutex` and `std::condition_variable`. Tasks are pushed onto the queue; idle threads pick them up.

A good default pool size is `std::thread::hardware_concurrency()` or a fixed cap like 8, since most IPC work is I/O-bound.

## Files

- `cpp/host/dispatch.cpp` (the `dispatch()` method, lines 141-251)
- `cpp/host/dispatch.hpp` (add pool members to the Dispatcher class)
